### PR TITLE
Added longley-rice model

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Adds Longley-Rice (ITM and ITWON) model.

- [x] Use map elevations
- [x] Expand map processing for ITM. Map database with default to `continentalTemperate`
- [x] Exposes LOS Simple model in CBA settings
- [x] Map, antenna and models cleanup

The following are moved to another PR.
~~Expand map processing to include clutter density~~
~~Use clutter for calculating signal strength~~
~~Make ITM calculations a bit more C++?~~

### CODE REVIEW

**Comment by TheMagnetar** [comment]
> We use `#pragma once` in other places.

I plan on changing this... in a cleanup pr

File: extensions/src/ACRE2Arma/signal/signal.hpp [hunk 1]
```diff
@@ -61,8 +65,12 @@ namespace acre {
         protected:
             uint32_t _debug_id;
             acre::signal::map_p _map;
-            acre::signal::model::multipath _signalProcessor_multipath;
-            acre::signal::model::Arcade    _signalProcessor_arcade;
+
+            acre::signal::model::Arcade      _signalProcessor_arcade;
```
Comment (jonpas) [inline extensions/src/ACRE2Arma/signal/signal.hpp:70]: Arcade the only one capitalized?

File: addons/sys_signal/fnc_getRadioClimateCode.sqf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Public: No
+ */
+
+params [["_map", ""]];
+
+if (_map == "") exitWith {MAP_CLIMATE_CONTINENTAL_TEMPERATE};
+
+_map = toLower _map;
+
+private _config = configProperties [configFile >> "CfgAcreSignal" >> "LongleyRiceRadioClimate", "isArray _x", true];
```
Comment (jonpas) [inline addons/sys_signal/fnc_getRadioClimateCode.sqf:22]: Do we want to support this config in the map mod itself?

File: addons/sys_signal/fnc_getRadioClimateCode.sqf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Example:
+ * ["Altis"] call acre_sys_signal_fnc_getRadioClimateCode
+ *
+ * Public: No
+ */
+
+params [["_map", ""]];
+
+if (_map == "") exitWith {MAP_CLIMATE_CONTINENTAL_TEMPERATE};
```
Comment (jonpas) [inline addons/sys_signal/fnc_getRadioClimateCode.sqf]: Where is `MAP_CLIMATE_CONTINENTAL_TEMPERATE` defined?

File: addons/sys_signal/CfgAcreSignal.hpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
+class CfgAcreSignal {
+    class LongleyRiceRadioClimate {
+        /*
+         * Definition of radio climates are the following:
+         *  - Equatiorial: Africa, along the equator
```
Comment (jonpas) [inline addons/sys_signal/CfgAcreSignal.hpp]:
```suggestion
         *  - Equatorial: Africa, along the equator
```

**Review by jonpas — CHANGES_REQUESTED** [review]

**Comment by jonpas** [comment]
Really neat!

File: addons/sys_signal/fnc_getRadioClimateCode.sqf (hunk 2 above)
Comment (TheMagnetar) [reply addons/sys_signal/fnc_getRadioClimateCode.sqf:22]: Better not...

File: extensions/src/ACRE2Arma/signal/signal.hpp (hunk 1 above)
Comment (TheMagnetar) [reply extensions/src/ACRE2Arma/signal/signal.hpp:70]: I will correct this in another PR (and rename it to casual). Let me have some feedback. Nevertheless, what convention do we take? Class names capital?

File: addons/sys_signal/fnc_getRadioClimateCode.sqf (hunk 3 above)
Comment (TheMagnetar) [reply addons/sys_signal/fnc_getRadioClimateCode.sqf]: Ooops... this check is also obsolete.

File: extensions/src/ACRE2Arma/signal/signal.hpp (hunk 1 above)
Comment (jonpas) [reply extensions/src/ACRE2Arma/signal/signal.hpp:70]: Coding guidelines don't specify exactly, but use `snake_case` in all examples, so I'd say non-capitalized.

**Review by jonpas — APPROVED** [review]
All issues addressed.

**Comment by TheMagnetar** [comment]
I have done some changes:

- Disable ITWOM for the moment
- Rename Arcade to Casual

If you want to have a look at it @jonpas

**Review by jonpas — DISMISSED** [review]
Looks good.

Maybe throw commit that disables ITWOM into a separate issue so it doesn't get lost and you have to search for the parts again.

Also, documentation? Is this considered experimental? Should write something somewhere either way. :)

**Comment by TheMagnetar** [comment]
OK, i will put experimental behind ITM. It's a good idea.

**Review by jonpas — APPROVED** [review]

**Pull request merged**